midend/interpreter.cpp

```diff
@@ -585,6 +585,56 @@ SymbolicVoid* SymbolicVoid::instance = new SymbolicVoid();
 
 /*****************************************************************************************/
 
+void ExpressionEvaluator::postorder(const IR::Operation_Ternary* expression) {
+    auto e0 = get(expression->e0);
+    if (e0->is<SymbolicError>()) {
+        set(expression, e0);
+        return;
+    }
+    auto e1 = get(expression->e1);
+    if (e1->is<SymbolicError>()) {
+        set(expression, e1);
+        return;
+    }
+    auto e2 = get(expression->e2);
+    if (e2->is<SymbolicError>()) {
+        set(expression, e2);
+        return;
+    }
+    auto clone = expression->clone();
+    BUG_CHECK(e0->is<SymbolicInteger>(), "%1%: expected an SymbolicInteger", e0);
```

Comment:
Shouldn't this be a SymbolicBool?

Reply:
I think that for Mux it should be SymbolicBool, but for Slice it could be SymbolicInteger. So, I made some changes to support SymbolicInteger and SymbolicBool. Now, I think that implementation is much better.